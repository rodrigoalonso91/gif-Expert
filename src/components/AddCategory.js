import PropTypes from 'prop-types'
import React from 'react'
import { useState } from 'react';

export const AddCategory = ({setCategory}) => {

    const placeholder = 'Buscar'
    const [inputValue, setInputValue] = useState('');

    const handleInputValue = (e) => {
        setInputValue(e.target.value);
    }

    const handleSubmit = (e) => {
        e.preventDefault();

        if(inputValue.trim() !== ''){
            setCategory( category => [inputValue,...category] );
            setInputValue('');
        }
    };

    return (
        <form onSubmit={handleSubmit}>
            <input
                type="text"
                placeholder={placeholder}
                value={inputValue}
                onChange={handleInputValue}
            />
        </form>
    )
}

AddCategory.propTypes = {
    setCategory: PropTypes.func.isRequired
}
