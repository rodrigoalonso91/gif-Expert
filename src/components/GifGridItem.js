import PropTypes from 'prop-types'
import React from 'react'

export const GifGridItem = ({title, url}) => {

    return (
	    <div className="card animate__animated animate__fadeIn">
            <img src={url} alt={title}></img>
            <p data-testid="p1">{title}</p>
        </div>
    )
};

GifGridItem.propTypes = {
	title : PropTypes.string.isRequired,
	url : PropTypes.string.isRequired
}
