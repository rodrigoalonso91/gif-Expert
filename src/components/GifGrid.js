import PropTypes from 'prop-types';
import { GifGridItem } from "./GifGridItem";
import { useFetchGifs } from '../hooks/useFetchGifs';

export const GifGrid = ({category, limit = 10}) => {

	const {data: images, loading} = useFetchGifs(category, limit)
 
    return (
		<>
			<h3 className="animate__animated animate__fadeIn">{category}</h3>	
			{loading && <p className="animate__animated animate__flash">Loading...</p>}
			<ol className='container'>
				{
					images.map( img  => (
						<GifGridItem
							key={img.id}
							{...img}
						/>
					))
				}
			</ol>
		</>
    )
}

GifGrid.propTypes = {
	category: PropTypes.string.isRequired,
}