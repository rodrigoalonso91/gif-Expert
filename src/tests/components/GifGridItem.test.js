import {render, screen} from '@testing-library/react'
import {GifGridItem} from '../../components/GifGridItem'

describe('Test in GifGridItem', () => { 

	const img = {
		id: 'sBikoeGRyOrK0',
		title: 'guitar shredding GIF',
		url: 'https://media0.giphy.com/media/sBikoeGRyOrK0/giphy.gif?cid=6c30809bl2bahk3vu4bhua3ulsjhiov792cbfylsl6eaukrx&rid=giphy.gif&ct=g'
	}

	test('should show the component correctly', () => {

		expect(render(<GifGridItem key={img.id} {...img}/>)).toMatchSnapshot();
		
	});

	test('should the title be in the p tag', () => { 

		render(<GifGridItem key={img.id} {...img}/>);

		const pTag = screen.getByTestId('p1');

		expect( pTag.textContent.trim() ).toBe(img.title)

	});

	test('should the img tag to have the src and alt values', () => { 

		render(<GifGridItem key={img.id} {...img}/>);

		const imgTag = screen.getByAltText(`${img.title}`);
		const srcValue = imgTag.getAttribute('src');
		const altValue = imgTag.getAttribute('alt');
		
		expect( srcValue ).toBe(img.url);
		expect( altValue ).toBe(img.title);

	});
});