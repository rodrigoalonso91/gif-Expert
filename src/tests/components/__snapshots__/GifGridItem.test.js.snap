// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test in GifGridItem should show the component correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="card animate__animated animate__fadeIn"
      >
        <img
          alt="guitar shredding GIF"
          src="https://media0.giphy.com/media/sBikoeGRyOrK0/giphy.gif?cid=6c30809bl2bahk3vu4bhua3ulsjhiov792cbfylsl6eaukrx&rid=giphy.gif&ct=g"
        />
        <p
          data-testid="p1"
        >
          guitar shredding GIF
        </p>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="card animate__animated animate__fadeIn"
    >
      <img
        alt="guitar shredding GIF"
        src="https://media0.giphy.com/media/sBikoeGRyOrK0/giphy.gif?cid=6c30809bl2bahk3vu4bhua3ulsjhiov792cbfylsl6eaukrx&rid=giphy.gif&ct=g"
      />
      <p
        data-testid="p1"
      >
        guitar shredding GIF
      </p>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
